editor/src/clj/editor/gui.clj

```diff
@@ -661,11 +746,190 @@
         :type type ; Explicitly include the type (pb-field is optional, so :type-box would be stripped otherwise).
         :child-index child-index))) ; Used to order sibling nodes in the SceneDesc.
 
+;; SDK api
+(defmacro layout-property-getter [prop-sym]
+  {:pre [(symbol? prop-sym)]}
+  (let [prop-kw (keyword prop-sym)]
+    `(g/fnk [~'prop->value ~prop-sym]
+       (get ~'prop->value ~prop-kw ~prop-sym))))
+
+(defn layout-property-set-fn [_evaluation-context node-id _old-value _new-value]
+  (concat
+    (g/invalidate-output node-id :prop->value)
+    (g/invalidate-output node-id :layout->prop->value)))
+
+;; SDK api
+(defmacro layout-property-setter [prop-sym]
+  {:pre [(symbol? prop-sym)]}
+  layout-property-set-fn)
```

Comment:
We don't actually need the `prop-sym` since they all use the same setter, but I added it for symmetry with `layout-property-getter`, and it means we can decide to use it if we need to in the future without updating `extension-spine` yet again.